Support new genome browser chain file format

claudiofr opened the pull request:
Closes #16539

Support new genome browser chain file format. A description of the format is found here:
https://genome.ucsc.edu/goldenPath/help/chain.html

Create a new chain.py file and Chain class to support the new datatype and add entries in datatypes_conf.xml.sample for the new datatype.

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

claudiofr committed:
Support genome browser chain file format.

Support a new datatype for the genome browser chain file format described here:

https://genome.ucsc.edu/goldenPath/help/chain.html

Add meta data which lists the number of chains in the file and a sniffer
for auto-detecting the chain format.

claudiofr committed:
Added more documentation

mvdbeek commented on lib/galaxy/datatypes/test/1.chain:
Both test files are too large, can you break this down into a smaller test dataset ? Does the second file test something that the first doesn't ? If not a single test is sufficient.

claudiofr replied:
Done

claudiofr committed:
Reduce number of test files from 2 to 1 and make it smaller.

Reduce number of test files from 2 to 1 and make it smaller.
Change doctest to be more comprehensive.
